### ISSUE TITLE
Spurious error on inline ASM

### ISSUE THREAD

**Issue by Coder-256** [rust-lang/rust-analyzer#18103]
**rust-analyzer version**: 0.3.2104-standalone

**rustc version**: 1.81.0 (eeb90cda1 2024-09-04)

**editor or extension**: VS Code extension v0.3.2104

**relevant settings**: N/A

**repository link (if public, optional)**: N/A

**code snippet to reproduce**: 
```rust
fn rdtscp() -> u64 {
    let hi: u64;
    let lo: u64;
    unsafe {
        std::arch::asm!(
            "rdtscp",
            out("rdx") hi,
            out("rax") lo,
            out("rcx") _,
            options(nomem, nostack, preserves_flags)
        );
    }
    (hi << 32) | lo
}
```

rust-analyzer shows an error on the line `out("rcx") _`:
```
`_` expressions may only appear on the left-hand side of an assignment
```

Screenshot:
<img width="705" alt="Screenshot 2024-09-11 at 11 51 29 PM" src="https://github.com/user-attachments/assets/a2b9ffed-1d59-435a-ba8b-615ef5569b16">
